### PR TITLE
Quick fix for SUPERTYPE_IS_EXTENSION_FUNCTION_TYPE

### PR DESCRIPTION
**Pull request by cypressious**
Fixes #KT-11876

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/quickfix/QuickFixRegistrar.kt [hunk 1]
```diff
@@ -371,6 +371,8 @@ class QuickFixRegistrar : QuickFixContributor {
 
         DATA_CLASS_NOT_PROPERTY_PARAMETER.registerFactory(AddValVarToConstructorParameterAction.QuickFixFactory)
 
-        NON_LOCAL_RETURN_NOT_ALLOWED.registerFactory(AddCrossInlineFix)
+        NON_LOCAL_RETURN_NOT_ALLOWED.registerFactory(ConvertExtensionToFunctionTypeFix)
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/QuickFixRegistrar.kt]: This seems to be an accidental change, please revert

Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/QuickFixRegistrar.kt]: Done

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.kotlin.types.KotlinType
+
+class ConvertExtensionToFunctionTypeFix(element: KtTypeReference, type: KotlinType) : KotlinQuickFixAction<KtTypeReference>(element) {
+
+    private val targetTypeStringShort = type.renderType(IdeDescriptorRenderers.SOURCE_CODE_SHORT_NAMES_IN_TYPES)
+    private val targetTypeStringLong = type.renderType(IdeDescriptorRenderers.SOURCE_CODE)
+
+    override fun getText() = "Convert supertype to '$targetTypeStringShort'"
+    override fun getFamilyName() = "Convert extension function type to regular function type."
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt]: No period here please

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val replaced = element.replaced(KtPsiFactory(project).createType(targetTypeStringLong))
+        ShortenReferences.DEFAULT.process(replaced)
+    }
+
+    private fun KotlinType.renderType(renderer: DescriptorRenderer) = buildString {
+        append('(')
+        arguments.dropLast(1).map { renderer.renderType(it.type) }.joinTo(this@buildString, ", ")
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt]: I'd add a test for more than one argument, and/or for type with type parameters

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt (hunk 2 above)
Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt]: Done

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt (hunk 3 above)
Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertExtensionToFunctionTypeFix.kt]: Done

**Comment by yole** [comment]
Thanks for your contribution!

**Pull request merged**